File: 프로그래머스/Lv.2/148653. 마법의 엘리베이터/마법의 엘리베이터.js
function solution(storey) {
    let count = 0
    
    while (storey) {
        let cur = storey % 10
        let next = parseInt(storey / 10) % 10
        
        if (cur > 5) {
            count += (10 - cur) // 5보다 크면 10까지 올라감
            storey += 10
        } else if (cur === 5) { // 5면 5번 누름
            count += cur
            storey += (next >= 5 ? 10 : 0) // 다음 층이 5보다 크면 10으로 올라가고, 아니면 내려감
        } else { // 5보다 작으면 0까지 내려감
            count += cur
        }
        
        storey = parseInt(storey / 10) // 자릿수 변경
    }
    
    return count
}

File: 프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.py
def solution(storey):
    answer = 0
    
    while storey > 0:
        digit = storey % 10 # 현재 자릿수
        
        if digit < 5: # 5 미만이면 그냥 이동
            answer += digit
        elif digit > 5: # 5 초과면 올림하여 더 작은 비용으로 이동
            answer += (10 - digit)
            storey += 10
        else: # 5인 경우 다음 자릿수를 고려해서 이동
            next_digit = (storey // 10) % 10
            if next_digit < 5:
                answer += digit
            else:
                answer += (10 - digit)
                storey += 10
        
        storey //= 10 # 다음 자릿수로 이동
    
    return answer